bears/c_languages/ArtisticStyleBear.py

```diff
@@ -0,0 +1,866 @@
+from coalib.bearlib.abstractions.Linter import linter
+from dependency_management.requirements.DistributionRequirement import (
+    DistributionRequirement)
+
+
+@linter(executable='astyle',
+        output_format='corrected',
+        use_stdin=True)
+class ArtisticStyleBear:
+    """
+    Artistic Style is a source code indenter, formatter,
+    and beautifier for the C, C++, C++/CLI, Objective‑C,
+    C# and Java programming languages.
+    """
+
+    LANGUAGES = {'C', 'C++', 'Objective-C', 'C#', 'Java'}
+    REQUIREMENTS = {DistributionRequirement(apt_get='astyle')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_FIX = {'Formatting'}
+    SEE_MORE = 'http://astyle.sourceforge.net/astyle.html'
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         bracket_style: str='',
+                         use_spaces: bool=None,
+                         indent_size: int=4,
+                         attach_braces_to_namespace: bool=True,
+                         attach_braces_to_class: bool=True,
+                         attach_braces_to_inline: bool=True,
+                         attach_braces_to_extern: bool=False,
+                         indent_classes: bool=True,
+                         indent_modifiers: bool=True,
+                         indent_switches: bool=True,
+                         indent_cases: bool=True,
+                         indent_namespaces: bool=False,
+                         indent_labels: bool=True,
+                         indent_preproc_block: bool=True,
+                         indent_preproc_definition: bool=True,
+                         indent_preproc_conditionals: bool=True,
+                         indent_column_one_comments: bool=True,
+                         pad_header_blocks: bool=True,
+                         pad_operators: bool=True,
+                         pad_parenthesis: bool=False,
+                         pad_parenthesis_out: bool=False,
+                         pad_parenthesis_in: bool=False,
+                         delete_empty_lines_in_func: bool=False,
+                         break_closing_braces: bool=False,
+                         break_elseifs: bool=False,
+                         break_one_line_headers: bool=False,
+                         add_braces_to_one_line_conditionals: bool=False,
+                         remove_braces_from_one_line_conditionals: bool=False,
+                         remove_comment_prefix: bool=True):
```

Comment:
I mean I have to provide an argument called as `astyle_config_file`, will do it in the review iteration. :wink: